tests/unit_test/unit-test-vpc.py

```diff
@@ -0,0 +1,150 @@
+        vpc_id = response["Vpc"]["VpcId"]
+
+        # Create a mock internet gateway (IGW)
+        response = ec2_client.create_internet_gateway()
+        igw_id = response["InternetGateway"]["InternetGatewayId"]
+
+        # Attach the IGW to the VPC
+        ec2_client.attach_internet_gateway(InternetGatewayId=igw_id, VpcId=vpc_id)
+
+        # Create a mock subnet
+        response = ec2_client.create_subnet(CidrBlock=vpc_cidr, VpcId=vpc_id)
+
+        # Perform your test assertions
+        self.assertEqual(response["Subnet"]["CidrBlock"], vpc_cidr)
+        self.assertEqual(response["Subnet"]["VpcId"], vpc_id)
+
+    @mock_ec2
+    def test_delete_vpc(self):
+        ec2_client = boto3.client("ec2", region_name=region)
+
+        # Create a mock VPC
+        response = ec2_client.create_vpc(CidrBlock=vpc_cidr)
+        vpc_id = response["Vpc"]["VpcId"]
+
+        # Delete the VPC
+        response = ec2_client.delete_vpc(VpcId=vpc_id)
+
+        # Test VPC deletion
+        self.assertEqual(response["ResponseMetadata"]["HTTPStatusCode"], 200)
+
+        # Try to retrieve the deleted VPC
+        with self.assertRaises(ec2_client.exceptions.ClientError) as context:
+            ec2_client.describe_vpcs(VpcIds=[vpc_id])
+
+        self.assertEqual(
+            context.exception.response["Error"]["Code"], "InvalidVpcID.NotFound"
+        )
+
+    @mock_ec2
+    def test_create_route_table(self):
+        # Create a mock VPC
+        ec2_client = boto3.client("ec2", region_name=region)
+        response = ec2_client.create_vpc(CidrBlock=vpc_cidr)
+        vpc_id = response["Vpc"]["VpcId"]
+        print(response)
+
+        # Create a mock route table
+        response = ec2_client.create_route_table(VpcId=vpc_id)
+        route_table_id = response["RouteTable"]["RouteTableId"]
+
+        # Perform your test assertions
+        self.assertEqual(response["RouteTable"]["VpcId"], vpc_id)
+        self.assertEqual(response["RouteTable"]["Associations"], [])
+        self.assertEqual(response["RouteTable"]["RouteTableId"], route_table_id)
+
+    @mock_ec2
+    def test_route_table_association(self):
+        # Create a mock VPC
+        ec2_client = boto3.client("ec2", region_name=region)
+        response = ec2_client.create_vpc(CidrBlock=vpc_cidr)
+        vpc_id = response["Vpc"]["VpcId"]
+
+        # Create a mock subnet
+        response = ec2_client.create_subnet(CidrBlock=vpc_cidr, VpcId=vpc_id)
+        subnet_id = response["Subnet"]["SubnetId"]
+
+        # Create a mock route table
+        response = ec2_client.create_route_table(VpcId=vpc_id)
+        route_table_id = response["RouteTable"]["RouteTableId"]
+
+        # Associate the route table with the subnet
+        response = ec2_client.associate_route_table(
+            RouteTableId=route_table_id, SubnetId=subnet_id
+        )
+        print(response)
+        # Perform your test assertions
+        self.assertEqual(response["AssociationId"], response["AssociationId"])
```

Comment:
here we need to compare it with some variable. It has same value twice

Reply:
Resolved in [d67f4bc](https://github.com/XgridInc/xc3/pull/18/commits/d67f4bcc757b95ad043c435e8c4acb4dd33c6a76)